src/Pos/Wallet/Web/ClientTypes.hs

```diff
@@ -131,16 +131,23 @@ mkCTxId = CTxId . CHash
 txIdToCTxId :: TxId -> CTxId
 txIdToCTxId = mkCTxId . sformat hashHexF
 
+convertTxOutputs :: [TxOut] -> [(CAddress, CCoin)]
+convertTxOutputs = map (addressToCAddress . txOutAddress &&& mkCCoin . txOutValue)
+
 mkCTx
     :: Address            -- ^ An address for which transaction info is forming
     -> ChainDifficulty    -- ^ Current chain difficulty (to get confirmations)
     -> TxHistoryEntry     -- ^ Tx history entry
     -> CTxMeta            -- ^ Transaction metadata
     -> CTx
-mkCTx addr diff THEntry {..} meta = CTx {..}
+mkCTx addr diff THEntry {..} meta =
+    -- placeholder, this functionality isn't implemented yet
+    let ctFrom = [] in
```

Comment:
Currently, the only way to retrieve transaction source addresses is via `getTxExtra`, which was initially created for an API endpoint in Explorer. There are two problems with it:

1. The relevant code is guarded by `#ifdef WITH_EXPLORER`
2. `getTxExtra` requires `MonadDB` constraint.

The former is a medium-size refactoring. The latter I don't know how to fix properly.

If `mkCTx` was monadic and had `MonadDB` constraint, the following code:

```haskell
do
    txExtra <- maybeThrow (Internal "Transaction is not in the DB") =<< getTxExtra _thTxId
    let ctFrom = convertTxOutputs $ map toaOut $ toList $ teInputOutputs txExtra
    pure CTx {..}
```

along with these imports:

```haskell
import           Pos.DB.Class           (MonadDB)
import           Pos.Txp.Core.Types     (TxOutAux (..))
import           Pos.Util               (maybeThrow)
import           Pos.Wallet.Web.Error   (WalletError (..))
import           Pos.Explorer           (TxExtra (..), getTxExtra)
```

would do the job.

I've put a placeholder for the time being.